Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2066,6 +2127,15 @@ namespace Babylon
                     for (const auto id : removed)
                     {
                         m_idToInputSource.erase(id);
+
+                        // remove the corresponding gampad if it exists
+                        for (const auto gamepadId : enabledGamepads)
+                        {
+                            if (id == gamepadId)
+                            {
+                                m_idToGamepadObject.erase(id);
+                            }
+                        }
```

Comment:
This block of logic is a bit peculiar. Without diving into the code very deeply, I'm not 100% sure what I think this should be, but this seems like a very odd way to remove an entry from a map. If the intention is just to remove the ID from `m_idToGamepadObject` if it exists, I don't think you need to check; [`std::map::erase`](https://en.cppreference.com/w/cpp/container/map/erase) can legally be given nonexistent keys to erase, in which case it will just do nothing and return 0. In that case, the only line you'd need of this block is the line that erases.

It's possible I'm missing a nuance, though. The presence of `enabledGamepads` here confuses me; is it indicating that we _only_ want to remove gamepads in this way if they were added this frame? How would they get removed if they were added on a different frame?

Reply:
you are right Justin, by the above logic i was trying to prevent removing a non-existent id. I will remove this then. Thank you!